state/txindex/kv/kv_test.go

```diff
@@ -212,23 +212,19 @@ func TestTxSearchEventMatch(t *testing.T) {
 			q:             "tx.height < 2 AND account.number = 3 AND account.number = 2 AND account.number = 5",
 			resultsLength: 0,
 		},
-		"Deduplication test - should return nothing if attribute repeats multiple times with match events": {
-			q:             "tx.height < 2 AND account.number = 3 AND account.number = 2 AND account.number = 5",
-			resultsLength: 0,
-		},
-		" Match range with match events": {
+		" Match range with special character": {
 			q:             "account.number < 2 AND account.owner = '/Ivan/.test'",
 			resultsLength: 0,
 		},
-		" Match range with match events 2": {
+		" Match range with special character 2": {
 			q:             "account.number <= 2 AND account.owner = '/Ivan/.test' AND tx.height > 0",
 			resultsLength: 1,
 		},
-		" Match range with match events contains with multiple items": {
+		" Match range with contains with multiple items": {
 			q:             "account.number <= 2 AND account.owner CONTAINS '/Iv' AND account.owner CONTAINS 'an' AND tx.height = 1",
 			resultsLength: 1,
 		},
-		" Match range with match events contains": {
```

Comment:
Are we testing all "if" branches in the changes to file `state/indexer/block/kv/util.go` above?

Reply:
I could not think of how to test the cases with UTs because those if branches are handling events indexed using old versions of the indexer. Unless we create different indexing options in our UTs. We could provide a custom indexing function to generate the event key differently and then test. That is basically what i did manually. And I did this with the backwards compatibility tests when running on top of data indexed with old versions.